blobstore/wants.go

```diff
@@ -3,52 +3,57 @@ package blobstore
 import (
 	"context"
 	"encoding/json"
+	"fmt"
 	"io"
 	"sync"
 
 	"github.com/cryptix/go/logging"
-	"github.com/davecgh/go-spew/spew"
+	"github.com/go-kit/kit/metrics/prometheus"
 	"github.com/pkg/errors"
-
 	"go.cryptoscope.co/luigi"
 	"go.cryptoscope.co/muxrpc"
+
 	"go.cryptoscope.co/ssb"
 )
 
-func dump(log logging.Interface, v interface{}, from string) {
-	var x interface{}
-	if msg, ok := v.(WantMsg); ok {
-		x = &msg
-	}
-
-	if msg, ok := v.(*WantMsg); ok {
-		m := make(map[string]int64)
-		for _, w := range *msg {
-			m[w.Ref.Ref()] = w.Dist
-		}
-		x = m
-	}
-
-	log.Log("from", from, "blobSpew", spew.Sprint(x))
-}
-
-func NewWantManager(log logging.Interface, bs ssb.BlobStore) ssb.WantManager {
+func NewWantManager(log logging.Interface, bs ssb.BlobStore, opts ...interface{}) ssb.WantManager {
 	wmgr := &wantManager{
 		bs:    bs,
 		wants: make(map[string]int64),
 		info:  log,
 	}
 
+	for i, o := range opts {
```

Comment:
Agreed in general but I wasn't sure if I want to keep them anyways... I'm still not sure about the best way to _partially inject_ these counters. It's really easy to wrap a whole interface but in some cases that isn't possible.

My biggest dislike is all the `if gauge != nil { .Add(1) }` lines which I want to place with functions at least before merging this. I'm all yours for better ideas.. :)

Reply:
Hmm, I think we'll have to discuss the partial injection in person, I don't really know what you mean by it. But I see what you mean about the conditional adds... I don't know the prometheus API enough to comment on that, but in general it would be nice not to have a hard dependency on prometheus... maybe adding hooks or something like that is a better approach?